app/controllers/treehouses_controller.rb

```diff
@@ -1,16 +1,21 @@
 class TreehousesController < ApplicationController
+  skip_before_action :authenticate_user!, only: :index
+
   def index
     @treehouses = policy_scope(Treehouse).order(created_at: :desc)
   end
 
   def new
     @treehouse = Treehouse.new
+    authorize @treehouse
   end
 
   def create
     @treehouse = Treehouse.new(treehouse_params)
+    authorize @treehouse
+    @treehouse.user = current_user
     if @treehouse.save
-      redirect_to treehouse_path(@treehouse)
+      redirect_to treehouses_path
```

Comment:
why not show the new created treehouse instead of all treehouses and redirect to treehouse_path(@treehouse)?